platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/Combinators.kt

```diff
@@ -365,6 +365,13 @@ fun Decoder.readStrings(): List<String> =
     }
 
 
+internal
+fun Decoder.readStringsSet(): Set<String> =
+    readCollectionInto({ size -> HashSet(size) }) {
```

Comment:
🤔 Are you cautious about ordering flakiness (I think if it matters, then List variant should be used) or about (de)serialization?

Reply:
> 🤔 Are you cautious about ordering flakiness (I think if it matters, then List variant should be used) or about (de)serialization?

Mostly about accidentally having a different order after deserialization, which can bring unnecessary differences and cache misses between CC and non-CC executions. I'd prefer such a low-level code to err on the side of caution.

Reply:
Understood. I'll change it to `LinkedHashSet`, it does not matter that much here anyway.

Long-term I think it would be nice to have a regular HashSet though with hash-code-changing tests to discover possible issues early, but it's way outside this PR.